Unable to unmarshal map[string][]byte with JSONPb

## 🐛 Bug Report

Unable to unmarshal map[string][]byte with JSONPb. Please see the code below

## To Reproduce

```
package main

import (
	"encoding/json"
	"fmt"
	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
	"google.golang.org/protobuf/encoding/protojson"
)

var (
	Marshaler = &runtime.JSONPb{
		protojson.MarshalOptions{
			Indent:        "    ",
			UseProtoNames: true,
		},
		protojson.UnmarshalOptions{
			DiscardUnknown: true,
		},
	}
)

func normal() {
	var m = make(map[string][]byte)
	var n = make(map[string][]byte)

	m["a"] = []byte("ttt")
	m["b"] = []byte("abc")

	b, err := json.Marshal(m)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(b))

	if err := json.Unmarshal(b, &n); err != nil {
		panic(err)
	}

	fmt.Println(n)
}

func proto() {
	var m = make(map[string][]byte)
	var n = make(map[string][]byte)

	m["a"] = []byte("ttt")
	m["b"] = []byte("abc")

	b, err := Marshaler.Marshal(m)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(b))

	if err := Marshaler.Unmarshal(b, &n); err != nil {
		panic(err)
	}

	fmt.Println(n)
}

func main() {
	normal()
	fmt.Println("=====")
	proto()
}
```


## Expected behavior

```
{"a":"dHR0","b":"YWJj"}
map[a:[116 116 116] b:[97 98 99]]
=====
{
    "a": "dHR0",
    "b": "YWJj"
}
map[a:[116 116 116] b:[97 98 99]]
```

## Actual Behavior

```
{"a":"dHR0","b":"YWJj"}
map[a:[116 116 116] b:[97 98 99]]
=====
{
    "a": "dHR0",
    "b": "YWJj"
}
panic: json: cannot unmarshal string into Go value of type []json.RawMessage
```

## Your Environment

```
go 1.17

require (
	github.com/grpc-ecosystem/grpc-gateway/v2 v2.10.0
	google.golang.org/protobuf v1.28.0
)

require (
	github.com/golang/protobuf v1.5.2 // indirect
	golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd // indirect
	golang.org/x/sys v0.0.0-20211216021012-1d35b9e2eb4e // indirect
	golang.org/x/text v0.3.7 // indirect
	google.golang.org/genproto v0.0.0-20220317150908-0efb43f6373e // indirect
	google.golang.org/grpc v1.45.0 // indirect
)
```

--- a/runtime/marshal_json_test.go
+++ b/runtime/marshal_json_test.go
@@ -197,6 +197,8 @@ var (
 		{data: "", json: `""`},
 		{data: proto.String(""), json: `""`},
 		{data: "foo", json: `"foo"`},
+		{data: []byte("foo"), json: `"Zm9v"`},
+		{data: []byte{}, json: `""`},
 		{data: proto.String("foo"), json: `"foo"`},
 		{data: int32(-1), json: "-1"},
 		{data: proto.Int32(-1), json: "-1"},

--- a/runtime/marshal_jsonpb.go
+++ b/runtime/marshal_jsonpb.go
@@ -280,6 +280,17 @@ func decodeNonProtoField(d *json.Decoder, unmarshaler protojson.UnmarshalOptions
 		return nil
 	}
 	if rv.Kind() == reflect.Slice {
+		if rv.Type().Elem().Kind() == reflect.Uint8 {
+			var sl []byte
+			if err := d.Decode(&sl); err != nil {
+				return err
+			}
+			if sl != nil {
+				rv.SetBytes(sl)
+			}
+			return nil
+		}
+
 		var sl []json.RawMessage
 		if err := d.Decode(&sl); err != nil {
 			return err

--- a/runtime/marshal_jsonpb_test.go
+++ b/runtime/marshal_jsonpb_test.go
@@ -523,6 +523,8 @@ var (
 		{data: uint64(1), json: "1"},
 		{data: proto.Uint64(1), json: "1"},
 		{data: "abc", json: `"abc"`},
+		{data: []byte("abc"), json: `"YWJj"`},
+		{data: []byte{}, json: `""`},
 		{data: proto.String("abc"), json: `"abc"`},
 		{data: float32(1.5), json: "1.5"},
 		{data: proto.Float32(1.5), json: "1.5"},

